src/main/java/de/doubleslash/keeptime/view/ProjectsListViewController.java

```diff
@@ -0,0 +1,395 @@
+
+            // TODO how to update currentProjectLabel when active project was edited?
+            realignProjectList();
+         });
+      });
+
+      // Add MenuItem to ContextMenu
+      contextMenu.getItems().addAll(changeWithTimeMenuItem, editMenuItem, deleteMenuItem);
+
+      return projectElement;
+   }
+
+   private Dialog<ButtonType> setUpDialogButtonType(final String title, final String headerText) {
+      final Dialog<ButtonType> dialog = new Dialog<>();
+      dialog.setTitle(title);
+      dialog.setHeaderText(headerText);
+      dialog.getDialogPane().getButtonTypes().addAll(ButtonType.OK, ButtonType.CANCEL);
+      return dialog;
+   }
+
+   private GridPane setUpEditProjectGridPane(final Project p) {
+      final GridPane grid = setUpGridPane(p.getName(), p.getColor(), p.isWork());
+
+      final Spinner<Integer> indexSpinner = new Spinner<>();
+      final int availableProjectAmount = model.getAvailableProjects().size();
+      indexSpinner.setValueFactory(new IntegerSpinnerValueFactory(0, availableProjectAmount - 1, p.getIndex()));
+      grid.add(indexSpinner, 1, 3);
+
+      return grid;
+   }
+
+   private void editProject(final ObservableList<Node> nodes, final Project p) {
+      final TextField projectNameTextField = (TextField) nodes.get(1);
+      final ColorPicker colorPicker = (ColorPicker) nodes.get(3);
+      final CheckBox isWorkCheckBox = (CheckBox) nodes.get(5);
+      final Spinner<Integer> indexSpinner = (Spinner<Integer>) nodes.get(7);
+      controller.editProject(p, projectNameTextField.getText(), colorPicker.getValue(), isWorkCheckBox.isSelected(),
+            indexSpinner.getValue());
+   }
+
+   public void tick() {
+      for (final Entry<Project, Label> entry : elapsedProjectTimeLabelMap.entrySet()) {
+         final Project p = entry.getKey();
+         final Label label = entry.getValue();
+
+         final long seconds = model.getPastWorkItems().stream().filter(work -> work.getProject().getId() == p.getId())
+               .mapToLong(work -> Duration.between(work.getStartTime(), work.getEndTime()).getSeconds()).sum();
+         label.setText(DateFormatter.secondsToHHMMSS(seconds));
+      }
+   }
+
+   public ListCell<Project> returnListCellOfProject() {
+      return new ListCell<Project>() {
+
+         @Override
+         protected void updateItem(final Project item, final boolean empty) {
+            super.updateItem(item, empty);
+
+            setText("");
+            if (item == null || empty) {
+               setGraphic(null);
+            } else {
+               final Pane graphic = (Pane) projectSelectionNodeMap.get(item);
+               if (graphic == null) {
+                  final Project addedProject = item;
+                  addProjectToProjectSelectionNodeMap(addedProject);
+               }
+               LOG.trace("Item: '{}' -> '{}'", item.getName(), projectSelectionNodeMap.get(item));
+               setGraphic(projectSelectionNodeMap.get(item));
```

Comment:
please use the same variable when logging. If you change one, the other will not change with it